hw/bitstream/BUILD

```diff
@@ -58,16 +67,67 @@ filegroup(
     srcs = select({
         "bitstream_skip": ["skip.bit"],
         "bitstream_vivado": ["//hw/bitstream/vivado:fpga_cw310_mask_rom"],
-        "bitstream_gcp": ["@bitstreams//:bitstream_mask_rom"],
+        "bitstream_gcp_splice": [":gcp_spliced_mask_rom"],
         "//conditions:default": ["@bitstreams//:bitstream_mask_rom"],
     }),
 )
 
-# TODO(lowRISC/opentitan#13603): Use `select` once we're uploading these
-# artifacts to GCP.
 filegroup(
     name = "mask_rom_otp_dev",
-    srcs = [
-        "//hw/bitstream/vivado:fpga_cw310_mask_rom_otp_dev",
-    ],
+    srcs = select({
+        "bitstream_skip": ["skip.bit"],
+        "bitstream_vivado": ["//hw/bitstream/vivado:fpga_cw310_mask_rom_otp_dev"],
+        "bitstream_gcp_splice": [":gcp_spliced_mask_rom_otp_dev"],
+        # FIXME(#13603) By default, this will actually do a local splice instead
+        # of retrieving the pre-spliced bitstream from the cache.
+        "//conditions:default": [":gcp_spliced_mask_rom_otp_dev"],
```

Comment:
@cfrantz I'm not sure what to do here. Initially, I wanted to pretend that this bitstream is present in the `@bitstreams` workspace, but that failed CI. I think what happened is it tried to build `:e2e_bootup_success_otp_dev` and, unsurprisingly, the target did not exist in `@bitstreams`.

So I thought I'd temporarily point the default case at `:gcp_spliced_mask_rom_otp_dev` until we start caching this bitstream. But that fails in CI because the runner doesn't have Vivado's `updatemem` command. https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=89081&view=logs&j=5df5adf8-3577-5927-f0b2-31afed2697ee&t=dcb8db2a-d944-5acc-e50e-bea93f2a14e9&l=4228

I guess maybe I should bring back the "manual" tag for `:e2e_bootup_success_otp_dev` until we start caching this particular bitstream.

Reply:
Agreed.

Reply:
Done. Thanks for your help!

Reply:
Aha, after chatting with @milesdai, I learned the `pool: FPGA` machines do not have Vivado installed.

I think I need to merge a preliminary PR that resolves the FIXME I added to this file. I need to pre-splice the dev OTP image into the Mask ROM bitstream, add it to the GCP cache, and then fix the default case for this `//hw/bitstream:mask_rom_otp_dev` target.

Reply:
Thinking about it more, I realized that the `:e2e_bootup_success_otp_dev` test shouldn't be built because we intentionally tagged it "manual". I think `run-fpga-cw310-tests.sh` is accidentally selecting it because it first does a fancy `rdeps` Bazel query, then builds those matching targets. Just took a shot at updating the script to respect the "manual" tag.